src/psqlgraph/hydrator.py

```diff
@@ -509,57 +514,117 @@ def is_parent_relation(self, label, relation):
         Bonus: Label could be added to all edges and will make the lookup faster.
 
         Args:
-            src_node: first node of the edge
-            dst_node: second node of the edge
-            edge_label: label of the edge
+            src_node: source node of the edge
+            dst_node: destination node of the edge
+            edge_label: identifier defined in the dictionary `links` section. It can be
+                the name of the edge, or the label
+            strict: raise error is edge is invalid
+        Raises:
+            PSQLGraphError if either no association is found or multiple associations are found
+            for the source and destination nodes. Specifying a label reduces the chances of finding
+            multiple associations.
         """
+        association_name = None
         if edge_label:
-            edge_class = self.models.Edge.get_subclass(edge_label)
-            if not edge_class:
-                logging.warning(f"Edge with label {edge_label} not found")
-            elif (
-                edge_class.__src_class__ == src_node.__class__.__name__
-                and edge_class.__dst_class__ == dst_node.__class__.__name__
-            ):
-                getattr(src_node, edge_class.__src_dst_assoc__).append(dst_node)
-                return
-            elif (
-                edge_class.__src_class__ == dst_node.__class__.__name__
-                and edge_class.__dst_class__ == src_node.__class__.__name__
-            ):
-                getattr(dst_node, edge_class.__src_dst_assoc__).append(src_node)
-                return
-            else:
-                logging.warning(
-                    "Edge with label {} is not allowed between nodes {} and {}".format(
-                        edge_label, src_node.label, dst_node.label
-                    )
-                )
+            # attempt to get association using link name - e.g., performed_on
+            association_name = self.get_association_by_edge_label(src_node, dst_node, edge_label)
 
-        link_found = False
-        for assoc_name, assoc_meta in src_node._pg_edges.items():
-            if isinstance(dst_node, assoc_meta["type"]):
-                getattr(src_node, assoc_name).append(dst_node)
-                link_found = True
-                break
-
-        if not link_found:
-            logging.debug(
-                "Could not find a direct relation between '{}'<->'{}'".format(
-                    src_node.label, dst_node.label
+        if association_name:
+            getattr(src_node, association_name).append(dst_node)
+            return
+
+        # attempt to get association by going through all links defined on the source node if needed.
+        association_names = self.get_association_by_edge_name(src_node, dst_node, edge_label)
+
+        if len(association_names) == 1:
+            getattr(src_node, association_names[0]).append(dst_node)
+
+        if len(association_names) > 1:
+            if strict:
```

Comment:
This looks a little funky with 3 if statements. They are all related, so perhaps if/elif/else.